src/MessageInput.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import Snackbar from '@material-ui/core/Snackbar';
+import Alert from '@material-ui/lab/Alert';
+
+const api = require('./api.js');
+
+const styles = theme => ({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  },
+  textField: {
+    width: '30%',
+    marginRight: 15
+  }
+});
+
+class MessageInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      id: `${this.props.id}`,
+      index: `${this.props.index}`,
+      prob: `${this.props.prob}`,
+      text: `${this.props.text}`,
+      success: false,
+      error: false,
+      warning: false
+    }
+  }
+
+  handleOnChange = event => {
+    console.log('Click');
+    console.log(event.target.value);
+  };
+
+  updateText = event => {
+    this.setState({
+      text: `${event.target.value}`
+    })
+  }
+
+  updateProb = event => {
+    this.setState({
+      prob: `${event.target.value}`
+    })
+  }
+
+  open = (result) => {
```

Comment:
Repetitive code as the `Gauge`-related component.  I would see if you can find a way to make them related so that code does not need to be copied and pasted.